### PR TITLE
Add support for setting the script of any npc

### PR DESCRIPTION
**Pull request by diamondo25**

### CODE REVIEW

File: src/channel_server/npc_packet.cpp [hunk 1]
```diff
@@ -164,6 +164,44 @@ PACKET_IMPL(npc_chat, int8_t type, game_map_object npc_id, const string &text, b
...
+		if (written_scripts == 0) {
+			current_packet
+				.add<packet_header>(SMSG_NPC_SET_SCRIPT)
+				.add<uint8_t>(0);
+		}
+		
+		current_packet
+			.add<game_map_object>(kvp.first)
+			.add<string>(kvp.second)
+			.add<int32_t>(20010101) // Start and end date
```
Comment (LazyBui) [inline src/channel_server/npc_packet.cpp]: I'd like to have a class that represents a date expression this stupid. Not required for this PR.

File: src/channel_server/npc_packet.cpp [hunk 2]
```diff
@@ -164,6 +164,44 @@ PACKET_IMPL(npc_chat, int8_t type, game_map_object npc_id, const string &text, b
...
 }
 
+auto npc_set_script(hash_map<int32_t, string> scripts) -> vector<packet_builder> {
+
+	vector<packet_builder> output;
+	
+	packet_builder current_packet;
+	uint8_t written_scripts = 0;
+
+	for (auto &kvp : scripts) {
```
Comment (LazyBui) [inline src/channel_server/npc_packet.cpp]: Can this be `const`?

File: src/channel_server/npc_packet.cpp [hunk 3]
```diff
@@ -164,6 +164,44 @@ PACKET_IMPL(npc_chat, int8_t type, game_map_object npc_id, const string &text, b
 	return builder;
 }
 
+auto npc_set_script(hash_map<int32_t, string> scripts) -> vector<packet_builder> {
```
Comment (LazyBui) [inline src/channel_server/npc_packet.cpp]: If `kvp` can be `const`, I would prefer this to be passed by `const` ref.

File: src/common/config/world.hpp [hunk 4]
```diff
@@ -250,6 +272,11 @@ namespace vana {
 			builder.add<string>(obj.scrolling_header);
 			builder.add<string>(obj.name);
 			builder.add<config::rates>(obj.rates);
+			builder.add<int32_t>(obj.npc_forced_script.size());
```
Comment (LazyBui) [inline src/common/config/world.hpp]: `uint32_t`

**Review by LazyBui — APPROVED** [review]

**Comment by diamondo25** [comment]
@LazyBui  Fixed comments

**Pull request merged**